### ISSUE TITLE
Add autocomplete to REPL

### ISSUE THREAD

**Issue by ziflex** [MontFerret/ferret#16]
It would improve usability of REPL if we added autocomplete for registered functions.

**Comment by SumLare** [comment]
I want to help with this one, not sure where to look up for registered functions

**Comment by ziflex** [comment]
@SumLare great!
For new, there is no such functionality, you need to add one, something like this to the compiler:

```go
func (c *Compiler) RegisteredFunctions() []string {
	res := make([]string, 0, len(c.functions))

	for name := range c.functions {
		res = append(res, name)
	}

	return res
}
```

**Comment by ziflex** [comment]
With syntax itself it's more tricky. You need to take a look at [the parser](https://github.com/MontFerret/ferret/tree/master/pkg/parser/fql)

**Comment by 3timeslazy** [comment]
@ziflex we can extract it from [this](https://github.com/MontFerret/ferret/blob/master/pkg/parser/fql/fql_lexer.go#L322)

**Comment by ziflex** [comment]
Closed #219
